demos/instr.cc

```diff
@@ -159,11 +155,36 @@ void UnwindStackAndSlowlyReturnTo(const void *address) {
       "mov x11, sp\n"
       "dc civac, x11\n"
       "dsb sy\n"
-      "isb\n"
       "ldr x30, [sp], #16\n"
-      "ret\n"::"r"(address));
+      "ret\n"::"r"(address - 1));
+#else
+#  error Unsupported CPU.
+#endif
+}
+
+#if !defined(__aarch64__)
+__attribute__((noinline))
+static void InvokedFunction(size_t *val) {
+#if defined(__x86_64__) || defined(_M_X64)
+  asm volatile("movq %%rbp, (%0)"::"r"(val));
+#elif defined(__i386__) || defined(_M_IX86)
+  asm volatile("movl %%ebp, (%0)"::"r"(val));
+#else
+#  error Unsupported CPU.
+#endif
+}
+
+bool OptimizedCompilation() {
```

Comment:
This function is obscure and fragile enough that I think we're better off without it. Whatever we were hoping to use it for, we should find a different way or decide not to do it.

Here, it looks like we're using it to early-exit if we were compiled without optimization. Instead, let's change the build so we *can't* be compiled without those optimizations.

I know the goal is to fix #26, where I was hoping we could behave sensibly under any optimization mode. I imagined we would identify the places where we depended on the compiler arranging things "just so", reduce that to a minimum, and then probably implement those parts in assembly so the compiler doesn't get a vote.

But if the only way we can make this program optimization-level-agnostic is to take *more* dependencies on the specifics of the compiler's behavior, that is a net loss and we should not proceed.